packages/openapi-v3/src/parameter-decorator.ts

```diff
@@ -46,263 +27,498 @@ const paramDecoratorStyle = Symbol('ParamDecoratorStyle');
-  export const body = function(name: string, schema?: SchemaObject) {
-    return param({name, in: 'body', schema});
+    binary: createParamShortCut(
+      'path',
+      typeAndFormatMap.binary.type,
+      typeAndFormatMap.binary.format,
+    ),
+    /**
+     * Define a parameter of "date" type that's read from request path.
+     * Usage: @param.path.date('paramName')
+     * 
+     * @param name Parameter name matching one of the placeholders in the path
+     */
+    date: createParamShortCut(
+      'path',
+      typeAndFormatMap.date.type,
+      typeAndFormatMap.date.format,
+    ),
+    /**
+     * Define a parameter of "dateTime" type that's read from request path.
+     * Usage: @param.path.dateTime('paramName')
+     * 
+     * @param name Parameter name matching one of the placeholders in the path
+     */
+    dateTime: createParamShortCut(
+      'path',
+      typeAndFormatMap.dateTime.type,
+      typeAndFormatMap.dateTime.format,
+    ),
+    /**
+     * Define a parameter of "password" type that's read from request path.
+     * Usage: @param.path.password('paramName')
+     * 
+     * @param name Parameter name matching one of the placeholders in the path
+     */
+    password: createParamShortCut(
+      'path',
+      typeAndFormatMap.password.type,
+      typeAndFormatMap.password.format,
+    ),
   };
 
   /**
-   * Define a parameter of `array` type
-   *
+   * Define a parameter of `array` type.
+   * 
    * @example
    * ```ts
-   * export class MyController {
-   *   @get('/greet')
-   *   greet(@param.array('names', 'query', 'string') names: string[]): string {
-   *     return `Hello, ${names}`;
-   *   }
+   * export class MyController {		
+   *   @get('/greet')		
+   *   greet(@param.array('names', 'query', 'string') names: string[]): string {		
+   *     return `Hello, ${names}`;		
+   *   }		
    * }
    * ```
-   * @param name Parameter name
-   * @param source Source of the parameter value
+   * 
+   * @param name Parameter name		
+   * @param source Source of the parameter value		
    * @param itemSpec Item type for the array or the full item object
    */
   export const array = function(
     name: string,
     source: ParameterLocation,
-    itemSpec: ItemType | ItemsObject,
+    itemSpec: SchemaObject | ReferenceObject,
   ) {
-    const items = typeof itemSpec === 'string' ? {type: itemSpec} : itemSpec;
-    if (source !== 'body') {
-      return param({name, in: source, type: 'array', items});
-    } else {
-      return param({name, in: source, schema: {type: 'array', items}});
-    }
+    return param({
+      name,
+      in: source,
+      schema: {type: 'array', items: itemSpec},
+    });
   };
 }
 
-function createParamShortcut(source: ParameterLocation, type: ParameterType) {
-  // TODO(bajtos) @param.IN.TYPE('foo', {required: true})
-  return (name: string) => {
-    return param({name, in: source, type});
-  };
+function createParamShortCut(
+  source: ParameterLocation,
+  type: string,
```

Comment:
Building on top of my previous comment, I think this argument should become:

```ts
typeOrSchema: string | {type: string, format: string})
```

(Preferably, use an existing OAI type/interface instead of `{type: string, format: string}`.)